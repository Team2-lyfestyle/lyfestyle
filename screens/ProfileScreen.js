import React from "react";
import { StyleSheet, Text, View, SafeAreaView, Image, ScrollView, TouchableHighlight} from "react-native";
import { Ionicons, MaterialIcons, AntDesign, FontAwesome, Feather } from '@expo/vector-icons'
import { LinearGradient } from 'expo-linear-gradient';

export default function Profile({navigation}) {
    return (
        <View style = {styles.container}>
        <ScrollView showsVerticalScrollIndicator = {false}>
            <View style = {styles.header}>
                <LinearGradient
                    colors={['rgba(rgba(0, 254, 212, 1))', 'transparent']}
                    style={{
                        position: 'absolute',
                        left: 0,
                        right: 0,
                        top: 0,
                        height: 300,
                    }}
                />
                <TouchableHighlight style = {styles.settingsIcon} onPress={() => navigation.navigate('Home')}>
                    <Feather name = "log-out" size = {23} color = "black"/>
                </TouchableHighlight>
                
                {/* Profile Image  */}
                <View style = {[styles.center, {marginTop: 8}]}>
                    <View style = {styles.profileImage}>
                        <Image source={require("../assets/images/profile_pic.jpg")} style = {styles.profileImage}></Image>
                    </View>
                </View>
                {/* Name */}
                <View style = {[styles.center, {marginTop: 30}, {marginBottom: 15}]}>
                    <Text style = {[{fontSize: 30}, { fontWeight: '400' }]}>John Doe</Text>
                    {/* Followers Count  */}
                    {/* <View style = {[{marginTop: 15}, {alignItems: 'center'}]}>
                        <Text style = {{fontSize: 15}}>150</Text>
                        <Text style = {{fontSize: 10}, {color: '#AEB5BC'}}>Followers</Text>
                    </View> */}
                </View>
            </View>
       
                {/* Bio  */}
                <View style = {styles.statsContainer}>
                    <View style = {[{marginTop: 10}]}>
                        <Text style = {[{fontSize: 15}, { fontWeight: 'bold' }]}>Bio </Text>
                    </View>
                </View>

                <View style = {styles.centerBio}>
                    <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla tincidunt enim quis laoreet dapibus. Nulla lacinia posuere diam aliquam tincidunt. Vivamus commodo ligula quis nisl placerat laoreet.</Text>
                </View>
                {/* Stats Container */}
                <View style = {styles.statsContainer}>
                    <View style = {[{marginTop: 10}]}>
                        <Text style = {[{fontSize: 15}, { fontWeight: 'bold' }]}>100 </Text>
                        <Text style = {[{fontSize: 10}, { fontWeight: 'bold' }, {color: 'grey'}]}>Posts</Text>
                    </View>
                </View>
                {/* Picture Container  */}
                <View style = {styles.pictureContainer}>
                    {/* Scroll pictures in the horizontal direction  */}
                    <ScrollView horizontal = {true} showsHorizontalScrollIndicator = {false}>
                        <View style = {styles.mediaImage}>
                            <Image source = {require("../assets/images/1.jpg")} style = {styles.image} resizeMode="cover"></Image>
                        </View>
                        <View style = {styles.mediaImage}>
                            <Image source = {require("../assets/images/2.jpg")} style = {styles.image} resizeMode="cover"></Image>
                        </View>
                        <View style = {styles.mediaImage}>
                            <Image source = {require("../assets/images/3.jpg")} style = {styles.image} resizeMode="cover"></Image>
                        </View>
                        <View style = {styles.mediaImage}>
                            <Image source = {require("../assets/images/4.jpg")} style = {styles.image} resizeMode="cover"></Image>
                        </View>
                        <View style = {styles.mediaImage}>
                            <Image source = {require("../assets/images/5.jpg")} style = {styles.image} resizeMode="cover"></Image>
                        </View>
                        <View style = {styles.mediaImage}>
                            <Image source = {require("../assets/images/6.jpg")} style = {styles.image} resizeMode="cover"></Image>
                        </View>
                        <View style = {styles.mediaImage}>
                            <Image source = {require("../assets/images/7.jpg")} style = {styles.image} resizeMode="cover"></Image>
                        </View>
                    </ScrollView>
                </View>
            </ScrollView>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#EFEdF4'
    },

    // Properties for Header container 
    header: {
        paddingTop: 15,
        paddingBottom: 16,
        backgroundColor: '#204051',
        // alignItems: 'center',
        // justifyContent: 'center',
        borderBottomWidth: 1,
        borderBottomColor: '#EBECF4',
        shadowColor: '#454D65',
        shadowOffset: { height: 5 },
        shadowRadius: 15,
        shadowOpacity: 0.2,
        zIndex: 10
    },
    
    center: {
        alignSelf: 'center',
    },

    // Properties for Settings icon 
    settingsIcon: {
        marginTop: 25,
        marginRight: 20,
        alignItems: 'flex-end',
        flex: 1
    },

    // Properties for the profile picture 
    profileImage: {
        width: 140,
        height: 140,
        borderRadius: 70,
        shadowColor: '#202020',
        shadowOffset: {width: 0, height: 0},
        shadowRadius: 5,
    },

    // Properties to center Bio for profile
    centerBio: {
        alignSelf: 'center',
        marginTop: 30,
        marginBottom: 20,
        marginLeft: 20,
        marginRight: 20,
    },

    // Stats = Number of posts container 
    statsContainer: {
        flexDirection: 'row',
        marginLeft: 20,
        marginRight: 20,
        borderBottomWidth: 1.5,
        paddingBottom: 5
        // alignSelf: 'center'
    },

    // Sets margin for image container 
    pictureContainer: {
        marginTop: 30,
        marginBottom: 30
    },

    // Container for images 
    mediaImage: {
        width: 170,
        height: 190,
        borderRadius: 12,
        overflow: 'hidden',
        marginHorizontal: 12.5
    },

    // Properties for images(posts) 
    image : {
        flex: 1,
        width: undefined,
        height: undefined
    }
})