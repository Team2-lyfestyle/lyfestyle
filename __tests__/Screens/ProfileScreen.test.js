import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProfileScreen from '../../screens/ProfileScreen';

import renderer from 'react-test-renderer';

describe('<ProfileScreen />', () => {
  it('Should render the <ProfileScreen /> component', () => {
    const wrapper = shallow(<ProfileScreen />);
    expect(wrapper.find(ProfileScreen)).toMatchSnapshot();
  });
});


test('renders correctly', () => {
  const tree = renderer.create(<ProfileScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
