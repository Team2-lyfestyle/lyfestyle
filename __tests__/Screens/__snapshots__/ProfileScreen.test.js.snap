// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EFEdF4",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#204051",
            "borderBottomColor": "#EBECF4",
            "borderBottomWidth": 1,
            "paddingBottom": 16,
            "paddingTop": 15,
            "shadowColor": "#454D65",
            "shadowOffset": Object {
              "height": 5,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 15,
            "zIndex": 10,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4278255316,
                0,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "height": 467,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            style={
              Object {
                "flex": 1,
                "height": 40,
                "marginLeft": 675,
                "marginRight": 22.5,
                "marginTop": 40,
                "opacity": 1,
                "width": 45,
              }
            }
          >
            <Text />
          </View>
        </RNGestureHandlerButton>
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
              },
              Object {
                "marginTop": 8,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderRadius": 100,
                "height": 213.5,
                "shadowColor": "#202020",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowRadius": 5,
                "width": 262.5,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "borderRadius": 100,
                  "height": 213.5,
                  "shadowColor": "#202020",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowRadius": 5,
                  "width": 262.5,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
              },
              Object {
                "marginTop": 30,
              },
              Object {
                "marginBottom": 15,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 30,
                },
                Object {
                  "fontWeight": "400",
                },
              ]
            }
          >
            John Doe
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1.5,
            "flexDirection": "row",
            "marginLeft": 22.5,
            "marginRight": 22.5,
            "paddingBottom": 5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 15,
                },
                Object {
                  "fontWeight": "700",
                },
              ]
            }
          >
            Bio 
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 40,
            "marginLeft": 22.5,
            "marginRight": 22.5,
            "marginTop": 40,
          }
        }
      >
        <Text>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla tincidunt enim quis laoreet dapibus. Nulla lacinia posuere diam aliquam tincidunt. Vivamus commodo ligula quis nisl placerat laoreet.
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1.5,
            "flexDirection": "row",
            "marginLeft": 22.5,
            "marginRight": 22.5,
            "paddingBottom": 5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 15,
                },
                Object {
                  "fontWeight": "700",
                },
              ]
            }
          >
            100 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 10,
                },
                Object {
                  "fontWeight": "700",
                },
                Object {
                  "color": "grey",
                },
              ]
            }
          >
            Posts
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 40,
            "marginTop": 40,
          }
        }
      >
        <RCTScrollView
          horizontal={true}
          showsHorizontalScrollIndicator={false}
        >
          <View>
            <View
              style={
                Object {
                  "borderRadius": 12,
                  "height": 333.5,
                  "marginHorizontal": 26.5,
                  "overflow": "hidden",
                  "width": 352.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "borderRadius": 12,
                  "height": 333.5,
                  "marginHorizontal": 26.5,
                  "overflow": "hidden",
                  "width": 352.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "borderRadius": 12,
                  "height": 333.5,
                  "marginHorizontal": 26.5,
                  "overflow": "hidden",
                  "width": 352.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "borderRadius": 12,
                  "height": 333.5,
                  "marginHorizontal": 26.5,
                  "overflow": "hidden",
                  "width": 352.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "borderRadius": 12,
                  "height": 333.5,
                  "marginHorizontal": 26.5,
                  "overflow": "hidden",
                  "width": 352.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "borderRadius": 12,
                  "height": 333.5,
                  "marginHorizontal": 26.5,
                  "overflow": "hidden",
                  "width": 352.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "borderRadius": 12,
                  "height": 333.5,
                  "marginHorizontal": 26.5,
                  "overflow": "hidden",
                  "width": 352.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
